### PR TITLE
Feature/gradle sbom

### PR DESCRIPTION
**Pull request by zulkhairi-edag**
Gradle file adapter

### CODE REVIEW

File: adapters/gradle/gradle.go [hunk 1]
```diff
@@ -1,8 +1,80 @@
...
+)
+
 type Gradle struct {
 }
 
 func (m *Gradle) Generate(file string) []byte {
+
+	InjectGradlePlugin(file)
+
+	_, err := utils.ExecCmd("gradle", "cyclonedxBom", "--build-file", "SBOMWorkingDir/build.gradle")
```
Comment (soon-yoong) [inline adapters/gradle/gradle.go:20]: the execCmd can be standardize to use shell from `structs/Shell.go` as per npm.go as the execCmd will be removed

File: adapters/gradle/gradle.go [hunk 2]
```diff
@@ -1,8 +1,80 @@
...
+
+	if err != nil {
+		console.Error(err.Error())
+	}
+
 	return nil
 }
+
+func IsPackage(resultInfo structs.ResultInfo) bool {
```
Comment (soon-yoong) [inline adapters/gradle/gradle.go:29]: this function has been removed from interface so I think we don't need this function anymore

File: adapters/gradle/gradle.go [hunk 3]
```diff
@@ -1,8 +1,80 @@
...
+	"sbom-tool/structs"
+	"sbom-tool/utils"
+)
+
 type Gradle struct {
 }
 
 func (m *Gradle) Generate(file string) []byte {
+
+	InjectGradlePlugin(file)
```
Comment (hazimhasnan-edag) [inline adapters/gradle/gradle.go:18]: better to add error to return from InjectGradlePlugin() and handle the error such as logging etc here

File: adapters/gradle/gradle.go (hunk 1 above)
Comment (zulkhairi-edag) [reply adapters/gradle/gradle.go]: updated the codes to use the new shell

**Review by dzakirin-muhammad — APPROVED** [review]

**Review by soon-yoong — APPROVED** [review]

**Pull request merged**